Blankmon/src/application/trainers/Trainer.java

```diff
@@ -0,0 +1,83 @@
+package application.trainers;
+
+import java.util.ArrayList;
+
+import application.Anature;
+import application.BaseAI;
+import application.Item;
+import application.enums.TrainerIds;
+import application.enums.Type;
+
+public class Trainer
+{
+	private TrainerIds mId;
+	private String mName;
+	private ArrayList<Anature> mAnatures;
+	private ArrayList<Item> mItems;
+	private Anature mCurrentAnature;
+	private BaseAI mAi;
+	private int mHealthThreshold;
+	private int mSwitchThreshold;
+
+	public Trainer(TrainerIds id, String name, ArrayList<Anature> anatures, ArrayList<Item> items, Anature currentAnature, BaseAI ai, int healthThreshold, int switchThreshold)
+	{
+		if(id == null && name == null && anatures == null && items == null && currentAnature == null && ai == null)
+			throw new IllegalArgumentException("Null Parameter");
+		
+		mId = id;
+		mName = name;
+		mAnatures = anatures;
+		mItems = items;
+		mCurrentAnature = currentAnature;
+		mAi = ai;
+		mHealthThreshold = healthThreshold;
+		mSwitchThreshold = switchThreshold;
+	}
+
+	public void switchAnature()
+	{
+		mAi.switchAnature(mAnatures, new Type[] { mCurrentAnature.getPrimaryType(), mCurrentAnature.getSecondaryType() }, 25, mCurrentAnature);
+	};
+
+	public final String useTurn(Anature playerAnature) // TODO The String return is only temporary
```

Comment:
Is this method just following the same logic that was built out in the BaseAI class?

Reply:
It's not supposed to, I just forgot to change it since switching wasn't gonna be apart of this demo. I was gonna create a hook method in the BaseAi class that would be called here instead. That way, if the switch is part of the turn it must pass the logic checks, but if its called by Trainer (meaning the controller called it because the trainer's anature fainted) then it can skip the logic and get straight to the switching.